src/main/java/org/apache/sysml/hops/Hop.java

```diff
@@ -77,8 +77,8 @@
 	protected long _nnz = -1;
 	protected UpdateType _updateType = UpdateType.COPY;
 
-	protected ArrayList<Hop> _parent = new ArrayList<Hop>();
-	protected ArrayList<Hop> _input = new ArrayList<Hop>();
+	protected ArrayList<Hop> _parent = new ArrayList<>(5);
+	protected ArrayList<Hop> _input = new ArrayList<>(5);
```

Comment:
Please leave the original specification for now and don't force the initial capacity - the array header anyway dominates here, so allocating an initial array of 5 instead of 10 elements makes no difference.